### PR TITLE
[+] More comprehensive docker guide

### PR DESCRIPTION
**Pull request by hykilpikonna**
* Added a section about automatically generating Dockerfile using ChatGPT 3.5-turbo
* Added guides on free automatic builds using Github Actions
* Added guides on deploying docker images and automatically updating the deployment

### CODE REVIEW

**Comment by terenceliu88** [comment]
Looks good!

**Review by terenceliu88 — APPROVED** [review]
Looks Good!

File: Topics/Development_Process/Docker.md [hunk 1]
```diff
@@ -13,45 +13,201 @@
...
 
 For detailed installation instructions based on specific operating systems click here: [Mac](https://docs.docker.com/desktop/install/mac-install/), [Windows](https://docs.docker.com/desktop/install/windows-install/), [Linux](https://docs.docker.com/desktop/install/linux-install/) 
 
-----
+## Creating Dockerfiles
 
-## Getting Started
-
-Once you've installed Docker, to see it in action you can follow any one of these quick tutorials:
+Once you've installed Docker, to see it in action you can follow any one of these quick tutorials on creating a Dockerfile that builds a docker image:
```
Comment (czhang0312) [inline Topics/Development_Process/Docker.md:25]: I think overall everything looks really good! Just one thing is that you explain how Docker packages applications into a container, but it's kind of unclear if a Dockerfile is the file that contains the containerized application. Also what is a docker image, is it just a Dockerfile? I think people who are new to Docker might also be confused by this, so maybe you should further clarify what each term means.

Comment (hykilpikonna) [reply Topics/Development_Process/Docker.md]: Fixed!

**Pull request merged**